From cb94741a98495cb139ec09da51a0e6106b3ab1be Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Mon, 29 Sep 2014 10:32:00 +0800
Subject: [PATCH 0246/1691] MLK-9618-3 usb: chipidea: otg: enable hnp_pollinig
 flag

commit cb94741a98495cb139ec09da51a0e6106b3ab1be from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

This patch enables hnp_polling flag for chipidea host driver in otg fsm mode.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit acb43a932263edd9b80c6d6d9ee858176c201542)
(cherry picked from commit 9e75e851370b62319fec4200b68cfd3465e12cfe)
---
 drivers/usb/chipidea/otg_fsm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/chipidea/otg_fsm.c b/drivers/usb/chipidea/otg_fsm.c
index cd58378..f5c63ba 100644
--- a/drivers/usb/chipidea/otg_fsm.c
+++ b/drivers/usb/chipidea/otg_fsm.c
@@ -898,6 +898,7 @@ int ci_hdrc_otg_fsm_init(struct ci_hdrc *ci)
 	ci->otg.gadget = &ci->gadget;
 	ci->fsm.otg = &ci->otg;
 	ci->fsm.power_up = 1;
+	ci->fsm.hnp_polling = 1;
 	ci->fsm.id = hw_read_otgsc(ci, OTGSC_ID) ? 1 : 0;
 	ci->fsm.otg->state = OTG_STATE_UNDEFINED;
 	ci->fsm.ops = &ci_otg_ops;
-- 
1.9.1

